### PR TITLE
[Gemma4] Refactor MMEncoderJITManager and add support to use it in Jax path Gemma4

### PR DESCRIPTION
**Pull request by kwang3939**
# Description

Based on the enable of using [Vit Cuda Graph](https://docs.vllm.ai/en/v0.22.0/design/cuda_graphs_multimodal/) on torchax path: 
PR: https://github.com/vllm-project/tpu-inference/pull/2804
Refactor PR: https://github.com/vllm-project/tpu-inference/pull/2802

Refactor the MMEncoderJITManager to be shared by both paths. Add adapter and funcs needed for jax path Gemma4 to use it.

# Tests

Verified the correctness of jax path Gemma4: `{'accuracy': 0.7116, 'correct': 1231, 'total': 1730, 'gen_num': 1730}`
Verified the correctness of torchax path `Qwen/Qwen3.5-397B-A17B-FP8`.
Tested with `python3 -m pytest -s -v tests/runner/test_mm_encoder_jit_manager.py`.

# Checklist

Before submitting this PR, please make sure:
- I have performed a self-review of my code.
- I have necessary comments in my code, particularly in hard-to-understand areas.
- I have made or will make corresponding changes to any relevant documentation.

### CODE REVIEW

File: tpu_inference/models/common/model_loader.py [hunk 1]
```diff
@@ -462,8 +464,26 @@ def wrapped_model_fn(*args, **kwargs):
         return jitted_model_fn(*args, **kwargs)
 
     compute_logits_fn = run_compute_logits
-    embed_multimodal_fn = run_embed_multimodal
     embed_input_ids_fn = run_embed_input_ids
+
+    _mm_jit = maybe_create_mm_encoder_jit_manager(
```
Comment (lk-chen) [inline tpu_inference/models/common/model_loader.py:469]: for Gemma4, how does it create manager? in `mm_encoder_jit_manager.py:451` it checks [support_encoder_cudagraph](https://github.com/vllm-project/vllm/blob/3d3b96488f5f7d94b1ab63919e0f1e7922a9ded6/vllm/model_executor/models/interfaces.py#L1703) which only checks if model is instance of `SupportsEncoderCudaGraph`. Gemma-4 is not yet inheriting `SupportsEncoderCudaGraph`?

Comment (kwang3939) [reply tpu_inference/models/common/model_loader.py:469]: For jax path I passed the jax model to check on `support_encoder_cudagraph`, so I added `supports_encoder_cudagraph = True` in our gemma4_mm.py.

Comment (lk-chen) [reply tpu_inference/models/common/model_loader.py:469]: how's this boolean used? vllm's `supports_encoder_cudagraph` method only checks `isinstance(..`

Comment (kwang3939) [reply tpu_inference/models/common/model_loader.py:469]: I had the same concern and tested it and found it was initialized correctly. I looked up a bit and found SupportsEncoderCudaGraph is a @runtime_checkable Protocol, so isinstance() just checks if supports_encoder_cudagraph exists as a class attribute, not actual inheritance.

File: tpu_inference/models/jax/gemma4_mm.py [hunk 2]
```diff
@@ -808,6 +812,218 @@ def embed_multimodal(self, **kwargs) -> List[jax.Array]:
...
+            modalities=["image"],
+            buffer_keys=["pixel_values", "pixel_position_ids"],
+            out_hidden_size=text_hidden,
+            max_frames_per_video=1,
+            padding_logics={"pixel_position_ids": pad_pixel_position_ids},
+        )
+        return config
+
+    def get_max_frames_per_video(self) -> int:
+        return 1  # images only
```
Comment (lk-chen) [inline tpu_inference/models/jax/gemma4_mm.py]: shall we raise unimplemented exception

File: tpu_inference/models/common/model_loader.py (hunk 1 above)
Comment (lk-chen) [reply tpu_inference/models/common/model_loader.py:469]: Got it, I was confused because vLLM models all do inheritance e.g. https://github.com/vllm-project/vllm/blob/dccb412e2c72a0c147166c14b25c01f045a74163/vllm/model_executor/models/deepseek_ocr.py#L366 If it's technically possible maybe we should do the same (inherit vllm class) for better readability.

File: tpu_inference/models/common/model_loader.py [hunk 3]
```diff
@@ -462,8 +464,26 @@ def wrapped_model_fn(*args, **kwargs):
...
+        precompile_vision_encoder_fn = _mm_jit.precompile_vision_encoder
+
+        def embed_multimodal_fn(state_leaves, modality=None, **kwargs):
+            if modality is not None and _mm_jit.supports_modality(modality):
+                return _mm_jit.execute(kwargs)
+            return run_embed_multimodal(state_leaves,
+                                        modality=modality,
+                                        **kwargs)
+    else:
+        embed_multimodal_fn = run_embed_multimodal
```
Comment (lk-chen) [inline tpu_inference/models/common/model_loader.py:487]: let's put a TODO here to implement this protocol for qwe2.5-vl and enforce all jax model to go this approach

**Review by lk-chen — APPROVED** [review]

File: tpu_inference/models/jax/gemma4_mm.py (hunk 2 above)
Comment (kwang3939) [reply tpu_inference/models/jax/gemma4_mm.py]: Done.

File: tpu_inference/models/common/model_loader.py (hunk 3 above)
Comment (kwang3939) [reply tpu_inference/models/common/model_loader.py:487]: Done.

File: tpu_inference/models/common/model_loader.py (hunk 1 above)
Comment (kwang3939) [reply tpu_inference/models/common/model_loader.py:469]: Just tried using explicit inheritance and found it causes a metaclass conflict. I'll keep this as it is then.

**Comment by lk-chen** [comment]
@copilot could you list per-file owner? don't at them, just put username

**Review by kyuyeunk — APPROVED** [review]
